tls/crypto.c

```diff
@@ -442,6 +457,25 @@ ttls_crypto_modinit(void)
 		}
 	}
 
+	/*
+	 * Below, so that when the "shaN" request is loaded, the driver with the
+	 * highest priority is loaded, even if some one with a lower priority is
+	 * already loaded, you must try to load all cripto kernel modules.
+	 */
+	for (inst_set = inst_sets; *inst_set; ++inst_set) {
+		for (h = ttls_hashes; h->info; ++h) {
+			size_t len = snprintf(NULL, 0, "crypto-%s%s",
+					      h->info->alg_name, *inst_set);
+			char *module = kmalloc(len + 1, GFP_KERNEL);
```

Comment:
You don't need to allocate memory here. `request_module()` is using `vsnprintf()` inside, so you can just:
```c
request_module("crypto-%s%s", h->info->alg_name, *inst_set);
```